kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/SchemaRegistryManager.java

```diff
@@ -0,0 +1,241 @@
+import io.streamnative.pulsar.handlers.kop.schemaregistry.model.SchemaStorageAccessor;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.model.impl.PulsarSchemaStorageAccessor;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.model.impl.SchemaStorageException;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.resources.CompatibilityResource;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.resources.ConfigResource;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.resources.SchemaResource;
+import io.streamnative.pulsar.handlers.kop.schemaregistry.resources.SubjectResource;
+import io.streamnative.pulsar.handlers.kop.security.KafkaPrincipal;
+import io.streamnative.pulsar.handlers.kop.security.auth.Authorizer;
+import io.streamnative.pulsar.handlers.kop.security.auth.Resource;
+import io.streamnative.pulsar.handlers.kop.security.auth.ResourceType;
+import io.streamnative.pulsar.handlers.kop.security.auth.SimpleAclAuthorizer;
+import io.streamnative.pulsar.handlers.kop.utils.MetadataUtils;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import javax.naming.AuthenticationException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.authentication.AuthenticationProvider;
+import org.apache.pulsar.broker.authentication.AuthenticationService;
+import org.apache.pulsar.broker.authentication.AuthenticationState;
+import org.apache.pulsar.broker.service.BrokerService;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.common.api.AuthData;
+import org.apache.pulsar.common.policies.data.ClusterData;
+
+@Slf4j
+public class SchemaRegistryManager {
+    private final KafkaServiceConfiguration kafkaConfig;
+    private final PulsarService pulsar;
+    private final AuthenticationService authenticationService;
+    private final SchemaRegistryRequestAuthenticator schemaRegistryRequestAuthenticator;
+    private final Authorizer authorizer;
+    private final PulsarClient pulsarClient;
+
+    public SchemaRegistryManager(KafkaServiceConfiguration kafkaConfig,
+                                 PulsarService pulsar,
+                                 AuthenticationService authenticationService) {
+        this.kafkaConfig = kafkaConfig;
+        this.pulsarClient = SystemTopicClient.createPulsarClient(pulsar, kafkaConfig, (___) -> {});
+        this.pulsar = pulsar;
+        this.authenticationService = authenticationService;
+        this.authorizer = new SimpleAclAuthorizer(pulsar);
+        this.schemaRegistryRequestAuthenticator = new HttpRequestAuthenticator(this.kafkaConfig,
+                this.authenticationService, this.authorizer);
+    }
```

Comment:
```suggestion
    private final SchemaRegistryRequestAuthenticator schemaRegistryRequestAuthenticator;
    private final PulsarClient pulsarClient;

    public SchemaRegistryManager(KafkaServiceConfiguration kafkaConfig,
                                 PulsarService pulsar,
                                 AuthenticationService authenticationService) {
        this.kafkaConfig = kafkaConfig;
        this.pulsarClient = SystemTopicClient.createPulsarClient(pulsar, kafkaConfig, (___) -> {});
        this.pulsar = pulsar;
        Authorizer authorizer = new SimpleAclAuthorizer(pulsar);
        this.schemaRegistryRequestAuthenticator = new HttpRequestAuthenticator(this.kafkaConfig,
                authenticationService, authorizer);
    }
```

[Codacy] Change `authenticationService` and `authorizer` to local variables.